core/aws-core/src/main/java/software/amazon/awssdk/awscore/endpoint/DefaultServiceEndpointBuilder.java

```diff
@@ -40,6 +43,7 @@
     private Region region;
     private ProfileFile profileFile;
     private String profileName;
+    private final Map<ServiceMetadataAdvancedOption<?>, Object> advancedOptions = new HashMap<>();
```

Comment:
Does AttributeMap.Builder work for this?

Reply:
It does, but the public API `ServiceMetadataConfiguration` would need to take AttributeMap or AttributeMap.Builder instead, which is not consistent with `ClientOverrideConfiguration#advancedOptions` 

https://github.com/aws/aws-sdk-java-v2/blob/040b7fde9da28c58b6d57cfad76b40de719369a9/core/regions/src/main/java/software/amazon/awssdk/regions/ServiceMetadataConfiguration.java#L116

https://github.com/aws/aws-sdk-java-v2/blob/040b7fde9da28c58b6d57cfad76b40de719369a9/core/sdk-core/src/main/java/software/amazon/awssdk/core/client/config/ClientOverrideConfiguration.java#L350